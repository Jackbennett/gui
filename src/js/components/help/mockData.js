export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
