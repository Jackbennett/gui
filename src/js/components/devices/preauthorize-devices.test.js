import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Preauthorize from './preauthorize-devices';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('PreauthorizeDevices Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      devices: {
        byStatus: {
          accepted: { total: 0 },
          preauthorized: { total: 0 }
        },
        filters: [],
        selectedDeviceList: [],
        limit: 500
      },
      users: {
        globalSettings: { id_attribute: null }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Preauthorize />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
