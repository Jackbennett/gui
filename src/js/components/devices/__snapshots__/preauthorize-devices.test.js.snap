// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreauthorizeDevices Component renders correctly 1`] = `
<div
  className="tab-container"
>
  <div
    className="flexbox space-between"
    style={
      Object {
        "marginBottom": -1,
        "zIndex": 2,
      }
    }
  >
    <div />
    <div
      className="flexbox centered"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Preauthorize devices
        </span>
      </button>
    </div>
  </div>
  <div
    className="loaderContainer"
  >
    <div
      className="loader"
    >
      <span
        className="dot dot_1"
      />
      <span
        className="dot dot_2"
      />
      <span
        className="dot dot_3"
      />
      <span
        className="dot dot_4"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <p>
      There are no preauthorized devices.
    </p>
    <p>
      <a
        onClick={[Function]}
      >
        Preauthorize devices
      </a>
       so that when they come online, they will connect to the server immediately
    </p>
    <img
      alt="preauthorize"
      src="preauthorize.png"
    />
  </div>
</div>
`;
